packages/dma-library/src/operations/ajna/open.ts

```diff
@@ -0,0 +1,158 @@
+import { actions } from '@dma-library/actions'
+import {
+  IOperation,
+  Protocol,
+  WithAjnaBucketPrice,
+  WithAjnaStrategyAddresses,
+  WithCollateral,
+  WithDebtAndBorrow,
+  WithFlashloan,
+  WithOptionalDeposit,
+  WithPosition,
+  WithProxy,
+  WithSwap,
+} from '@dma-library/types'
+import { NULL_ADDRESS, ZERO } from '@oasisdex/dma-common/constants'
+import { ajnaOpenOperationDefinition } from '@oasisdex/dma-deployments/operation-definitions'
+import BigNumber from 'bignumber.js'
+import { ethers } from 'ethers'
+
+type OpenArgs = WithCollateral &
+  WithDebtAndBorrow &
+  WithOptionalDeposit &
+  WithSwap &
+  WithFlashloan &
+  WithProxy &
+  WithPosition &
+  WithAjnaStrategyAddresses &
+  WithAjnaBucketPrice
+
+export async function open({
+  collateral,
+  debt,
+  deposit,
+  swap,
+  flashloan,
+  proxy,
+  position,
+  addresses,
+  price,
+}: OpenArgs): Promise<IOperation> {
+  const depositAmount = deposit?.amount || ZERO
+  const depositAddress = deposit?.address || NULL_ADDRESS
+
+  const isDepositingCollateral = deposit?.address === collateral.address
+  const isDepositingDebtTokens = deposit?.address === debt.address
+
+  const pullDebtTokensToProxy = actions.common.pullToken({
+    asset: debt.address,
+    amount: isDepositingDebtTokens ? depositAmount : ZERO,
+    from: proxy.owner,
+  })
```

Comment:
it all comes to question whether we need to allow users to deposit also debt tokens. On the designs we allow only depositing collateral tokens.

Reply:
not necessary as we just pull collateral.